src/base/Notifier.sol

```diff
@@ -35,10 +34,14 @@ abstract contract Notifier is INotifier {
         if (_subscriber != NO_SUBSCRIBER) revert AlreadySubscribed(address(_subscriber));
         subscriber[tokenId] = ISubscriber(newSubscriber);
 
-        try ISubscriber(newSubscriber).notifySubscribe(tokenId, config, data) {}
-        catch (bytes memory reason) {
-            revert Wrap__SubsciptionReverted(newSubscriber, reason);
+        bool success = _call(
+            address(newSubscriber), abi.encodeWithSelector(ISubscriber.notifySubscribe.selector, tokenId, config, data)
+        );
+
+        if (!success) {
+            Wrap__SubsciptionReverted.selector.bubbleUpAndRevertWith(address(newSubscriber));
         }
```

Comment:
just my opinion -- but i dont how i feel about the manual call handler just so we can standardize around `.bubbleUpAndRevertWith`

like it feels less readable now

Reply:
yeahhh i get that